mlir/include/mlir/IR/MLIRContext.h

```diff
@@ -281,6 +303,9 @@ class MLIRContext {
   }
 
 private:
+  /// Set the remark engine for this context.
+  void setRemarkEngine(std::unique_ptr<remark::detail::RemarkEngine> engine);
```

Comment:
Just like `enableOptimizationRemarksToFile`, I believe `enableOptimizationRemarks` does not belong to the context and should be a method of the `RemarkEngine` or be a free function in `Remarks.h` instead.
I don't think the context should be a factory for the RemarkEngine here.